quinn-proto/src/connection/mod.rs

```diff
@@ -3618,6 +3612,89 @@ impl fmt::Debug for Connection {
     }
 }
 
+/// Fields of `Connection` specific to it being client-side or server-side
+enum ConnectionSide {
+    Client {
+        /// Sent in every outgoing Initial packet. Always empty after Initial keys are discarded
+        token: Bytes,
+    },
+    Server {
+        server_config: Arc<ServerConfig>,
+    },
+}
+
+impl ConnectionSide {
+    fn side(&self) -> Side {
+        match *self {
+            Self::Client { .. } => Side::Client,
+            Self::Server { .. } => Side::Server,
+        }
+    }
+
+    fn is_client(&self) -> bool {
+        self.side().is_client()
+    }
+
+    fn is_server(&self) -> bool {
+        self.side().is_server()
+    }
+
+    fn migration(&self) -> bool {
```

Comment:
This name seems to be the inverse of its behavior and use. Also, `is_` for predicates.

Reply:
Good point. Inverted both the method and its usage, and gave more descriptive name:

```diff
diff --git a/quinn-proto/src/connection/mod.rs b/quinn-proto/src/connection/mod.rs
index 1f2e803f..6dc4157e 100644
--- a/quinn-proto/src/connection/mod.rs
+++ b/quinn-proto/src/connection/mod.rs
@@ -1074,7 +1074,7 @@ impl Connection {
                 // If this packet could initiate a migration and we're a client or a server that
                 // forbids migration, drop the datagram. This could be relaxed to heuristically
                 // permit NAT-rebinding-like migration.
-                if remote != self.path.remote && self.side.migration() {
+                if remote != self.path.remote && !self.side.remote_may_migrate() {
                     trace!("discarding packet from unrecognized peer {}", remote);
                     return;
                 }
@@ -3647,10 +3647,10 @@ impl ConnectionSide {
         self.side().is_server()
     }
 
-    fn migration(&self) -> bool {
+    fn remote_may_migrate(&self) -> bool {
         match self {
-            ConnectionSide::Server { server_config } => !server_config.migration,
-            ConnectionSide::Client { .. } => true,
+            ConnectionSide::Server { server_config } => server_config.migration,
+            ConnectionSide::Client { .. } => false,
         }
     }
 }
```